pilot/pkg/bootstrap/monitoring.go

```diff
@@ -36,13 +38,30 @@ const (
 	versionPath = "/version"
 )
 
+var (
+	uptime      = monitoring.NewGauge("istiod_uptime_seconds", "Current istiod server uptime in seconds")
+	serverStart time.Time
+)
+
+func init() {
+	monitoring.MustRegister(uptime)
+	serverStart = time.Now()
+}
+
+func exportUptime(h http.Handler) http.Handler {
```

Comment:
is this the recommended way to compute a metric during scrape time?

Reply:
Seems like we could use NewGaugeFunc?

Reply:
I was trying to avoid using Prom-specifc methods. The istio.io/pkg/monitoring package doesn't have a NewGaugeFunc. I don't think this metric is problematic, however.

Reply:
@howardjohn wdyt about the new approach?  it will require additions to the istio.io/pkg/monitoring pkg, but doesn't hold us to a single implementation.